modules/smoothxg/main.nf

```diff
@@ -0,0 +1,35 @@
+def VERSION = '0.6.1' // Version information not provided by tool on CLI
+
+process SMOOTHXG {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? 'bioconda::smoothxg=0.6.1' : null)
+
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/smoothxg:0.6.1--h735f0e5_0' :
+        'quay.io/biocontainers/smoothxg:0.6.1--h735f0e5_0' }"
+
+    input:
+    tuple val(meta), path(gfa)
+
+    output:
+    tuple val(meta), path("*.gfa"), emit: gfa
+    path "versions.yml"           , emit: versions
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    smoothxg \\
```

Comment:
I am voting for a more complex script, which ensures we can run several iterations of `smoothxg`: https://github.com/nf-core/pangenome/blob/571d649719db87ffc7769eb0b3f526df99cf1e7a/main.nf#L269-L343. 
It might be possible to port some Bash magic to Groovy, but I doubt it will be much.

Reply:
I seem to recall a new Nextflow feature that might help with this, will look into it

Reply:
https://github.com/nextflow-io/patterns/blob/master/docs/feedback-loop.adoc

Reply:
It was also discussed in https://github.com/nextflow-io/nextflow/discussions/2521#discussioncomment-2374495.

So the idea would be to code the whole thing in a subworkflow. Which means we leave this module as it is?

Reply:
What is still unclear to me is if the output of the previous iteration really is the input of the current iteration using the recursion feature.